src/plugins/console/server/lib/proxy_request.ts

```diff
@@ -77,8 +81,39 @@ export const proxyRequest = ({
   });
 
   req.once('response', (res) => {
-    resolved = true;
-    resolve(res);
+    // Check if the request is to _mapping endpoint and if so, limit the response to 10MB. This is to
+    // protect against large mapping responses that can cause the browser to hang.
+    if (isMappingEndpoint(pathname)) {
```

Comment:
Please see documentation 

<img width="717" alt="image" src="https://user-images.githubusercontent.com/20072247/194523246-59b93648-0b06-4b9d-89a9-ec509ab41987.png">

Your code would make sense if callback works with partial data, in the current implementation it looks strange

@elastic/kibana-core do we have a legal way to limit request by payload size?

Reply:
Sorry, I gonna need slightly more context here. What kind of request are we talking about (or are we talking about response here) ?

Requests to our endpoints? For payload-capable requests, the `options.body.maxBytes` (RouteConfigOptionsBody.maxBytes) can be used.

Requests performed from Kibana toward ES? No AFAIK we don't currently have an out-of-the-box way of doing it.

Reply:
Thanks for your reply @pgayvallet! We are talking about the response from es here. The problem we are trying to solve is that sometimes the response is too large and the browser hangs. This tends to happen for users with a large number of mappings. We are trying to find a way to limit the response size. What do you think about this  [approach](https://github.com/elastic/kibana/pull/140569/commits/3b75822ea4a569cf5578119d4da9564e24c1fd03#diff-dc97f758bd78991f73f2f928fc59d00923790286e12766dd7c91d148ec40cfeeR50) I'm not sure if there is a better way to do this.

Reply:
First question: I know the console proxy requests are not using the ES library client, but are you forced to use an low level`http` client here instead of using the ES client exposed from Core given the ES endpoint you're reaching it not dynamic? If you were using the ES client lib to do your requests, you could use the `TransportRequestOptions.maxResponseSize` option to do what you want.


Now if you do need to use the same approach as other console ES calls, your approach looks fine. One question though: if the issue is about sending the response back to the client, do you really need to stream the raw response from ES to consume the size on each data emissions? Can't you just wait for the whole response and then look at it size to decide if you should be throwing or forwarding the response? It's less optimized for sure, but the implementation would be simpler.

Reply:
Only reason we're using the low-level http client is that the ES client doesn't seem to support limiting the response size. I've tried using the `maxResponseSize` option, but it doesn't seem to work. I've also tried using the `maxCompressedResponseSize` option, but that doesn't seem to work either. I could not find any use cases of these options in the codebase, except for the tests. I'm not sure if it's a bug or if I'm using it incorrectly.
                 
It makes sense to wait for the whole response and then look at its size. But would that be a problem if the response is very large? I'm not sure if that would cause a memory issue. What do you think? @pgayvallet @sabarasaba @alexwizp